### PR TITLE
Add followedBy combinator

### PR DESCRIPTION
**Pull request by LukaJCB**
Should address #39 
Willing to bikeshed the naming :P

### CODE REVIEW

**Comment by cb372** [comment]
Let the bikeshedding commence! How about `andThen` or `followedBy`?

**Comment by LukaJCB** [comment]
I'm fine with both of these, with a slight preference to `followedBy` :)

**Comment by cb372** [comment]
Ok let’s go with followedBy

File: modules/core/shared/src/main/scala/retry/RetryPolicy.scala [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+    RetryPolicy(status =>
+      M.map2(decideNextRetry(status), rp.decideNextRetry(status)) {
+        case (GiveUp, pd) => pd
+        case (pd, _)      => pd
+    })
+}
 
 object RetryPolicy {
 
+  def alwaysGiveUp[M[_]: Applicative]: RetryPolicy[M] =
```
Comment (cb372) [inline modules/core/shared/src/main/scala/retry/RetryPolicy.scala]: Let's move this to the `RetryPolicies` object

**Comment by cb372** [comment]
Looks good apart from my comment. Just needs a final rebase.

**Pull request merged**